torch/export/__init__.py

```diff
@@ -0,0 +1,279 @@
+from typing import Any, Callable, Dict, List, Optional, Tuple
+
+import torch
+
+
+__all__ = [
+    "dynamic_dim",
+    "export",
+]
+
+def dynamic_dim(t: torch.Tensor, index: int):
```

Comment:
@avikchaudhuri please take a look at this part. Ignore the `export()` part for now, it is being reviewed in a different PR, my ghstack is acting up, so I can't push proper stacks now.